torch/csrc/jit/codegen/cuda/runtime/tensor_core.cu

```diff
@@ -0,0 +1,410 @@
+
```

Comment:
For all uses of `nvfuser_index_t`, do they need to be that type? Can any of them be safely replaced with just `int`? For example, indexing inside a `32x8` tile should be safe with `int`.

Reply:
Yes I will replace them all with `int` when we get to a point when these swizzling can be done outside of main loop.